### ISSUE TITLE
kic: clearify image build process for kic

### ISSUE THREAD

**Issue by medyagh** [kubernetes/minikube#6301]
for kic the minikube docker-env command wont work.
we need to provide a good clear story for the users, how to use their images.

minikube cache add is one way
we need to explore the registry addons we have to see if they work as intended.

**Comment by afbjorklund** [comment]
I think running with containerd has similar issues (with building images) ?
But at least this driver (kic) has a local docker daemon to build them with...

The main problem is how to transport the newly built images to minikube.
One approach would be to save them to stdout, and load them from stdin:

`docker save | sudo ctr --namespace=k8s.io images import -`

But it doesn't seem like it is working, with the current `ssh` connection.
At least not without saving it to a file, and using `docker cp` or `scp`.

Microk8s has the same problem: https://microk8s.io/docs/registry-images

Basically it is a "known issue", when using kubernetes with containerd.

**Comment by afbjorklund** [comment]
Minikube main alternatives for building images:
- Use the docker daemon (for container runtime `docker`)
- Use the podman service (for container runtime `cri-o`)

Workarounds, for other simpler environments:
- Build locally, push to the machine (for `containerd`)
- Build locally, push to the registry (for [Registry 2.x](https://github.com/docker/distribution))

Eventually it should be possible to build images _inside_ the Kubernetes cluster natively, but currently that will just fallback to one of the above methods. But one can for instance run rootless Buildah in a pod...

For the short term it would probably be best to offer a straight-forward (one-time) `load` function, and perhaps that `build` wrapper that will for instance call the local `docker` or `podman-remote` too ?

**Comment by tstromberg** [comment]
Image builds for kic should work exactly like other platforms.

Closing, since this doesn't seem to be going anywhere.
